internal/webpagecapture/webpagecapture.go

```diff
@@ -0,0 +1,56 @@
+package webpagecapture
+
+import (
+	"context"
+	"crypto/tls"
+	"io"
+	"net/http"
+
+	webscan "github.com/Method-Security/webscan/generated/go"
+)
+
+// PerformWebpageCapture performs a webpage capture against a target URL
+func PerformWebpageCapture(ctx context.Context, target string) *webscan.WebpageCaptureReport {
+	report := &webscan.WebpageCaptureReport{
+		Target: target,
+		Errors: []string{},
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+
+	resp, err := client.Get(target)
+	if err != nil {
+		report.Errors = append(report.Errors, err.Error())
+		return report
+	}
+	defer func() {
+		if cerr := resp.Body.Close(); cerr != nil {
+			err = cerr
+		}
+	}()
+	if err != nil {
+		report.Errors = append(report.Errors, err.Error())
+		return report
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		report.Errors = append(report.Errors, "Failed to get the webpage. Status code: "+string(rune(resp.StatusCode)))
+		return report
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		report.Errors = append(report.Errors, err.Error())
+	} else {
+		bodyString := string(body)
+		report.Html = &bodyString
```

Comment:
I would suggest that we Base64 this when we send it via the Signal

Reply:
like so it gets double base64 encoded?

Reply:
so the output json is {"html": "<base64 enocded strin>"} ?